src/cmd-prune-containers

```diff
@@ -0,0 +1,168 @@
+        case "w" | "W":
+            days = value * 7
+        case "d" | "D":
+            days = value
+        case _:
+            raise ValueError(f"Invalid unit '{match.group(2)}'. Please use y (years), m (months), w (weeks), or d (days).")
+
+    return days
+
+
+def skopeo_delete(repo, image, auth):
+
+    skopeo_args = ["skopeo", "delete", f"docker://{repo}:{image}"]
+    if auth is not None:
+        skopeo_args.append(f"--authfile {auth}")
+
+    subprocess.run(skopeo_args)
+
+
+# FIXME : move to cosa_lib
+# https://github.com/coreos/coreos-assembler/pull/3798/files#r1673481990
+
+def parse_fcos_version(version):
+    m = regexp.match(r'^([0-9]{2})\.([0-9]{8})\.([0-9]+)\.([0-9]+)$', version)
+    if m is None:
+        raise ValueError(f"Incorrect versioning for FCOS build {version}")
+    try:
+        timestamp = datetime.datetime.strptime(m.group(2), '%Y%m%d')
+    except ValueError:
+        raise Exception(f"FCOS build {version} has incorrect date format. It should be in (%Y%m%d)")
+    return (timestamp, int(m.group(3)))
+
+
+def get_update_graph(stream):
+
+    url = f"https://raw.githubusercontent.com/coreos/fedora-coreos-streams/main/updates/{stream}.json"
+    r = requests.get(url)
+    if r.status_code != 200:
+        raise Exception(f"Could not download update graph for {stream}. HTTP {r.status_code}")
+    return r.json()
+
+
+class BarrierRelease(Exception):
+    pass
+
+
+def main():
+
+    args = parse_args()
+
+    # Load the policy file
+    with open(args.policy, "r") as f:
+        policy = yaml.safe_load(f)
+    if args.stream not in policy:
+        raise Exception(f"Stream {args.stream} is not defined in policy file.")
+    if 'containers' not in policy[args.stream]:
+        raise Exception(f"No containers section for {args.stream} stream in policy.")
+    policy = policy[args.stream]["containers"]
+
+    print(f"Pulling tags from {args.repository_url}")
+    # This is a JSON object:
+    # {"Repository": "quay.io/jbtrystramtestimages/fcos",
+    #    "Tags": [
+    #        "40.20"40.20240301.1.0",.....]}
+    tags_cmd = subprocess.run(["skopeo", "list-tags",
+                              f"docker://{args.repository_url}"],
+                              capture_output=True)
+    tags_data = tags_cmd.stdout
+    if tags_cmd.returncode != 0:
+        raise Exception(f"Error retrieving tags list from repo: {tags_cmd.stdout}")
+
+    tags_json = json.loads(tags_data)
+    tags = tags_json['Tags']
+    # Compute the date before we should prune images
+    # today - prune-policy
+    today = datetime.datetime.now()
+    date_limit = today - relativedelta(days=convert_to_days(policy))
+    if args.v:
+        print(f"This will delete any images older than {date_limit} from the stream {args.stream}")
+
+    stream_id = STREAMS[args.stream]
+    # Get the update graph for stable streams
+    if stream_id in [1, 2, 3]:
+        update_graph = get_update_graph(args.stream)['releases']
+
+    for tag in tags:
+
```

Comment:
Let's just here do

```python
if tag in STREAMS:
    continue
...
```

?

I know this would get caught by the `except ValueError` below, but it feels odd to treat tags we _know_ will not be conformant as "unexpected".

That way it makes the "unexpected" log message a better signal we should pay attention to.

Reply:
I had done exactly this before in the bash version then you wrote last week : https://github.com/coreos/coreos-assembler/pull/3826#discussion_r1665043870

So.... :) Maybe I can tweak the error message ?

Reply:
Hmm, I don't see them as conflicting. :) The bash version was too permissive, and now this one is too restrictive. Anyway, not a strong opinion!

Reply:
Ok I get what you mean now. Now we'll only have "unexpted" logs on tags that are truely not supposed to be there :)